packages/map-template/src/components/QRCodeDialog/QRCodeDialog.jsx

```diff
@@ -0,0 +1,117 @@
+import primaryColorState from "../../atoms/primaryColorState";
+import apiKeyState from "../../atoms/apiKeyState";
+import currentLocationState from "../../atoms/currentLocationState";
+import QRCode from 'qrcode';
+import kioskLocationState from "../../atoms/kioskLocationState";
+import logoState from "../../atoms/logoState";
+
+/**
+ * Handle the QR Code dialog.
+ *
+ */
+function QRCodeDialog() {
+    const [, setShowQRCodeDialog] = useRecoilState(showQRCodeDialogState);
+
+    const primaryColorProp = useRecoilValue(primaryColorState);
+    const apiKeyProp = useRecoilValue(apiKeyState);
+    const logoProp = useRecoilValue(logoState);
+
+    const directionsFrom = useRecoilValue(kioskLocationState);
+    const directionsTo = useRecoilValue(currentLocationState);
+
+    useEffect(() => {
+        if (directionsFrom && directionsTo) {
+            // The current URL when the user opens the QR code dialog
+            let currentUrl = window.location.origin;
+
+            // The interface for the existing URL search params
+            const currentParams = new URLSearchParams(window.location.search);
+
+            // The interface for the new URL search params
+            const newParams = new URLSearchParams();
+
+            /**
+             * Function that handles the presence of query parameters
+             * and appends them to the newParams interface.
+             * 
+             * @param {string} queryParam 
+             * @param {string} prop 
+             */
+            function handleQueryParams(queryParam) {
+                if (currentParams.has(queryParam)) {
+                    const queryParameter = currentParams.get(queryParam);
+                    newParams.append(queryParam, queryParameter);
+                }
+            }
+
+            /**
+             * Function that handles the presence of query parameters and props
+             * and appends them to the newParams interface.
+             * 
+             * @param {string} queryParam 
+             * @param {string} prop 
+             */
+            function handleQueryParamsAndProps(queryParam, prop) {
+                if (currentParams.has(queryParam)) {
+                    const queryParameter = currentParams.get(queryParam);
+                    newParams.append(queryParam, queryParameter);
+                } else if (prop) {
+                    if (prop === primaryColorProp) {
+                        newParams.append(queryParam, primaryColorProp.replace("#", ""))
+                    } else {
+                        newParams.append(queryParam, prop)
+                    }
+                }
+            }
+
+            // Handle query parameters for the gmApiKey and mapboxAccessToken
+            handleQueryParams('gmApiKey');
+            handleQueryParams('mapboxAccessToken');
+
+            // Handle query parameters and props for apiKey, primaryColor, logo 
+            handleQueryParamsAndProps('apiKey', apiKeyProp);
```

Comment:
Can you explain why these three props should be handled differently from the two above?

Reply:
Because in the case of apiKey, primaryColor and logo we need to have the else statement which checks for the existence of the prop. 
My thought was that in the case of gmApiKey and mapboxAccessToken we don't need to do that, as one of them will always be present, therefore we only need to check the query params. 
Do you think it would make sense to handle all the props the same as we do for the apiKey, primaryColor and logo?

Reply:
Yeah, wouldn't it be more simple to handle it in the same way?

Reply:
Yes, that's true. Refactored the code to respect that ✅